### ISSUE TITLE
Incorrectly closed websockets are throwing unobserved task exceptions 2

### ISSUE THREAD

**Issue by JustArchi** [dotnet/corefx#26830]
This issue is very similar to https://github.com/dotnet/corefx/issues/24002 but with entirely different stacktrace.

Basically, I have a simple code:

```c#
try {
	HttpListenerWebSocketContext webSocketContext = await context.AcceptWebSocketAsync(null).ConfigureAwait(false);

	while (webSocketContext.WebSocket.State == WebSocketState.Open) {
		WebSocketReceiveResult result = await webSocketContext.WebSocket.ReceiveAsync(new byte[0], CancellationToken.None).ConfigureAwait(false);

		if (result.MessageType != WebSocketMessageType.Close) {
			await webSocketContext.WebSocket.CloseAsync(WebSocketCloseStatus.InvalidMessageType, "You're not supposed to be sending any message but Close!", CancellationToken.None);
			break;
		}

		await webSocketContext.WebSocket.CloseAsync(WebSocketCloseStatus.NormalClosure, string.Empty, CancellationToken.None).ConfigureAwait(false);
		break;
	}
} catch (Exception e) {
	Console.WriteLine(e);
}
```

Now, this code works totally fine in normal operation. However, there seems to be some kind of race condition if remote client sends `close` frame, but won't wait for connection being fully closed before abandoning it:

```
System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Es wurde versucht, auf eine nicht vorhandene Netzwerkverbindung zuzugreifen) ---> System.Net.HttpListenerException: Es wurde versucht, auf eine nicht vorhandene Netzwerkverbindung zuzugreifen (translation: An attempt was made to access a non-existent network connection)
   at System.Net.WebSockets.WebSocketHttpListenerDuplexStream.WriteAsyncFast(HttpListenerAsyncEventArgs eventArgs)
   at System.Net.WebSockets.WebSocketHttpListenerDuplexStream.<CloseNetworkConnectionAsync>d__50.MoveNext()
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.Net.HttpListenerException (0x80004005): Es wurde versucht, auf eine nicht vorhandene Netzwerkverbindung zuzugreifen (translation: An attempt was made to access a non-existent network connection)
   at System.Net.WebSockets.WebSocketHttpListenerDuplexStream.WriteAsyncFast(HttpListenerAsyncEventArgs eventArgs)
   at System.Net.WebSockets.WebSocketHttpListenerDuplexStream.<CloseNetworkConnectionAsync>d__50.MoveNext()<---
```

Now, the problem is that this is task exception being thrown deep in corefx code, and it's unobserved, so it gets thrown back on `TaskScheduler.UnobservedTaskException` which I use for debugging (and catching incidents like this one).

I believe this exception should be observed in .NET Core code, in similar way like done in #24002. I didn't find any way to solve this issue from consumer perspective, since the exception is thrown entirely out of my program's stacktrace, so it's uncatchable (as you can see from my code sample, and the stacktrace itself).

Thank you in advance for looking into this. I'm sorry if this issue was already reported, I couldn't find anything related despite of searching, and my original issue #24002 doesn't seem to cover this scenario (unless I'm wrong, please correct me if I am). Cheers!

[EDIT] Add C# syntax highlighting by @karelz

**Comment by karelz** [comment]
Can you please try the code on 2.1? See https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/dogfooding.md
Bug #24002 was fixed in 2.1.

**Comment by JustArchi** [comment]
I actually managed to reproduce this bug also on master, but luckily while reproducing it I accidentally found the real reason for this - build that doesn't go through `ILLink.Tasks` step works properly. This proved that the bug I'm dealing with has nothing to do with corefx, so I'm sorry for taking a bit of your time, and thank you deeply for help.

I'll report it to mono linker devs instead 🙂
